python/hail/keytable.py

```diff
@@ -257,8 +257,11 @@ def join(self, right, how='inner'):
          - **left** -- Key present in ``kt1``. For keys only in ``kt1``, the value of non-key columns from ``kt2`` is set to missing.
          - **right** -- Key present in ``kt2``. For keys only in ``kt2``, the value of non-key columns from ``kt1`` is set to missing.
 
-        .. note::
-            Both KeyTables must have identical key schemas and non-overlapping column names.
+        The non-key fields in ``kt2`` must have non-overlapping column names with ``kt1``.
+
+        Both KeyTables must have the same number of keys and the corresponding types of each key must be the same (order matters), but the key names can be different.
```

Comment:
I believe we've decided that in the docs we should call VDSes "variant dataset" and KTs "key table" instead of "VariantDataset" and "KeyTable" (unless linking)